docs/develop/contracts/rust/wasm-vmcontext-and-env.md

```diff
@@ -0,0 +1,22 @@
+---
+id: wasm-vmcontext-and-env
+title: Understanding the Importance of WebAssembly, VMContext, and env
+sidebar_label: Wasm, VMContext, and env
+---
+
+# Understanding the Importance of WebAssembly, VMContext, and env
+## WebAssembly
+NEAR compiles smart contracts to [WebAssembly](https://webassembly.org/), or Wasm. Wasm helps NEAR protocol transactions to be fast, safe, and open.
+
+TODO
+
+## VMContext
+TODO
+
+## env
+[`near-sdk::env`](https://docs.rs/near-sdk/latest/near_sdk/env/index.html) makes blockchain specific methods available to the smart contract. These methods include (but aren't limited to):
+* [`account_balance`](https://docs.rs/near-sdk/latest/near_sdk/env/fn.account_balance.html)
+* [`current_account_id`](https://docs.rs/near-sdk/latest/near_sdk/env/fn.current_account_id.html)
+* [`is_valid_account_id`](https://docs.rs/near-sdk/latest/near_sdk/env/fn.is_valid_account_id.html)
```

Comment:
why do we want to advertise this? If anything we should note that AccountId's are being validated through its type (as of v4) and not require using this API which is very unergonomic and should only be used in edge cases